src/dbgshim/debugshim.cpp

```diff
@@ -766,43 +759,61 @@ HRESULT CLRDebuggingImpl::FormatLongDacModuleName(_Inout_updates_z_(cchBuffer) W
 #endif
 
     const WCHAR* pDacBaseName = NULL;
-    if(m_skuId == CLR_ID_V4_DESKTOP)
+    if (m_skuId == CLR_ID_V4_DESKTOP)
         pDacBaseName = CLR_DAC_MODULE_NAME_W;
-    else if(m_skuId == CLR_ID_CORECLR || m_skuId == CLR_ID_PHONE_CLR || m_skuId == CLR_ID_ONECORE_CLR)
+    else if (m_skuId == CLR_ID_CORECLR || m_skuId == CLR_ID_PHONE_CLR || m_skuId == CLR_ID_ONECORE_CLR)
         pDacBaseName = CORECLR_DAC_MODULE_NAME_W;
     else
     {
         _ASSERTE(!"Unknown SKU id");
         return E_UNEXPECTED;
     }
 
+    _ASSERTE(targetImageFileMachine != IMAGE_FILE_MACHINE_ARM64X);
+
     const WCHAR* pTargetArch = NULL;
-    if(targetImageFileMachine == IMAGE_FILE_MACHINE_I386)
+    if (targetImageFileMachine == IMAGE_FILE_MACHINE_I386)
     {
         pTargetArch = W("x86");
     }
-    else if(targetImageFileMachine == IMAGE_FILE_MACHINE_AMD64)
+    else if (targetImageFileMachine == IMAGE_FILE_MACHINE_AMD64
+#ifdef HOST_AMD64
+             || targetImageFileMachine == IMAGE_FILE_MACHINE_ARM64EC
+#endif
+        )
     {
         pTargetArch = W("amd64");
     }
-    else if(targetImageFileMachine == IMAGE_FILE_MACHINE_ARMNT)
+    else if (targetImageFileMachine == IMAGE_FILE_MACHINE_ARMNT)
     {
         pTargetArch = W("arm");
     }
-    else if(targetImageFileMachine == IMAGE_FILE_MACHINE_ARM64)
+    else if (targetImageFileMachine == IMAGE_FILE_MACHINE_ARM64
+#ifdef HOST_ARM64
+             || targetImageFileMachine == IMAGE_FILE_MACHINE_ARM64EC
```

Comment:
targetImageFileMachine should never be IMAGE_FILE_MACHINE_ARM64EC -> it's always x64 or arm64. We don't do special compilations of the runtime